common/web/lm-worker/src/main/model-compositor.ts

```diff
@@ -166,7 +108,14 @@ export default class ModelCompositor {
     // Section 1:  determine 'prediction roots' - enumerate corrections from most to least likely,
     // searching for results that yield viable predictions from the model.
 
-    const { postContextState, rawPredictions } = await this.correctAndEnumerate(transformDistribution, context);
+    const SEARCH_TIMEOUT = correction.SearchSpace.DEFAULT_ALLOTTED_CORRECTION_TIME_INTERVAL;
+    const timer = this.activeTimer = new correction.ExecutionTimer(this.testMode ? Number.MAX_VALUE : SEARCH_TIMEOUT, this.testMode ? Number.MAX_VALUE : SEARCH_TIMEOUT * 1.5);
+
+    const { postContextState, rawPredictions } = await correctAndEnumerate(this.contextTracker, this.lexicalModel, timer, transformDistribution, context);
+
+    if(this.activeTimer == timer) {
+      this.activeTimer = null;
+    }
```

Comment:
Construction and state-management related to the `ExecutionTimer` used during correction-search was moved out of `correctAndEnumerate` in order to ensure `correctAndEnumerate` would be stateless.

If the goal is to be stateless, we can't have the method managing the "prediction process is active" `ModelCompositor` state signal.  This _does_ start the timer a bit earlier than usual, but the cost should easily be less than 1ms in practice; the parts between the two locations are essentially constant-time setup overhead.